lib/legacy-loader.mjs

```diff
@@ -5,16 +5,14 @@ import { send } from './ipc.mjs';
 const require = createRequire(import.meta.url);
 
 export async function getFormat(url, context, defaultGetFormat) {
-  const getPackageType = require('get-package-type');
   const filePath = fileURLToPath(url);
 
   send({ required: filePath });
+
   try {
     return await defaultGetFormat(url, context, defaultGetFormat);
-  } catch (err) {
-    if (err.code === 'ERR_UNKNOWN_FILE_EXTENSION') {
-      return { format: await getPackageType(filePath) };
-    }
-    throw err;
+  } catch (error) {
+    if (error.code !== 'ERR_UNKNOWN_FILE_EXTENSION') throw error;
+    return require('get-package-type')(filePath).then(format => ({ format, source: null }));
```

Comment:
This shouldn't break anything, but `getFormat`'s return value should only have the `format` key.